### PR TITLE
[Android] Fix Color range mapping in MediaCodec

### PR DESCRIPTION
**Pull request by joseluismarti**
## Description

MediaCodec is configured incorrectly as FFmpeg and MediaFormat use different values to identify the color ranges:

|| Limited | Full
|---|:-:|:-:|
|[FFmpeg](https://github.com/FFmpeg/FFmpeg/blob/891ed24f77da99c6d41bb7c116ba5925e3206ce2/libavutil/pixfmt.h#L626)|1|2
|[MediaFormat](https://developer.android.com/reference/android/media/MediaFormat#COLOR_RANGE_FULL)|2|1

This PR corrects the mapping and uses names instead of values to avoid confusion.

Depends on https://github.com/xbmc/libandroidjni/pull/45

## Types of change
- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp [hunk 1]
```diff
@@ -1392,7 +1392,20 @@ bool CDVDVideoCodecAndroidMediaCodec::ConfigureMediaCodec(void)
...
+      {
+        case AVCOL_RANGE_MPEG:
+          mediaformat.setInteger(CJNIMediaFormat::KEY_COLOR_RANGE,
+                                 CJNIMediaFormat::COLOR_RANGE_LIMITED);
+          break;
+        case AVCOL_RANGE_JPEG:
+          mediaformat.setInteger(CJNIMediaFormat::KEY_COLOR_RANGE,
+                                 CJNIMediaFormat::COLOR_RANGE_FULL);
+          break;
+        default:;
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp:1406]:
```c++
default:
  break;
```

... to be clear what the intention is and to align with our coding guidelines.

**Review by ksooo — COMMENTED** [review]
Could this be backported?

**Comment by joseluismarti** [comment]
I can do the backport including the library bump and test if there are any issues, although changes in the library are usually new methods/constants and should be fine.

Another option is to avoid the bump and exceptionally use the value of the two constants.

**Comment by thexai** [comment]
Media files are always limited range, but probably many times FFmpeg reports "unspecified". In other words, this has nothing to do with using FULL or LIMITED range for the video output. It only indicates that the source video file is limited/full range.

Probably the bug could only occur in the hypothetical case of a full range video (in case there is one exist).

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp [hunk 2]
```diff
@@ -1392,7 +1392,25 @@ bool CDVDVideoCodecAndroidMediaCodec::ConfigureMediaCodec(void)
   if (CJNIBase::GetSDKVersion() >= 24)
   {
     if (m_hints.colorRange != AVCOL_RANGE_UNSPECIFIED)
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp:1394]: Why don't you handle this case in the switch case statement below?

Comment (joseluismarti) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp:1394]: For consistency with the existing code I followed the handling logic used with the other video flags and it seemed to me to be fine. If the value is "Unspecified" we avoid directly doing the case condition checks.

Comment (ksooo) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp:1394]: If you put the "unspecified" as first case there should be no overhead in the generated code, just one comparism. But with the code as it is now, if the value is not "unspecified" you will do at least two comparisons, which IS overhead.

Comment (joseluismarti) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp:1394]: I know, as I said I wanted to use the same style as the similar code in the method. I'll change it as you suggest if we see it better

Comment (ksooo) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp]: Thanks.

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp [hunk 3]
```diff
@@ -1391,8 +1391,25 @@ bool CDVDVideoCodecAndroidMediaCodec::ConfigureMediaCodec(void)
 
   if (CJNIBase::GetSDKVersion() >= 24)
   {
-    if (m_hints.colorRange != AVCOL_RANGE_UNSPECIFIED)
-      mediaformat.setInteger(CJNIMediaFormat::KEY_COLOR_RANGE, m_hints.colorRange);
+    switch (m_hints.colorRange)
+    {
+      case AVCOL_RANGE_UNSPECIFIED:
+        CLog::Log(LOGWARNING, "CDVDVideoCodecAndroidMediaCodec::ConfigureMediaCodec Color range: "
```
Comment (thexai) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp:1397]: This is probably the most common for 90% of media files. For this reason I think it makes little sense to issue a WARNING in log  (unspecified --> Do nothing , all is good)

Comment (joseluismarti) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp:1397]: In the tests I did of starting about 30 videos (mostly HEVC, some AVC and one VP9) only one had the unspecified value.

Although I do not see any problem in changing the log level to DEBUG in this case.

Comment (thexai) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp]: This is much more frequent in older videos: H.264, VC1, MPEG2...

**Review by thexai — APPROVED** [review]

**Pull request merged**